ui/apps/platform/src/Containers/Dashboard/PatternFly/Widgets/ViolationsByPolicyCategory.tsx

```diff
@@ -37,19 +40,29 @@ import { getRequestQueryStringForSearchFilter } from 'utils/searchUtils';
 import useURLSearch from 'hooks/useURLSearch';
 import useSelectToggle from 'hooks/patternfly/useSelectToggle';
 import LIFECYCLE_STAGES from 'constants/lifecycleStages';
-import { LifecycleStage } from 'types/policy.proto';
+import { LifecycleStage, policySeverities, PolicySeverity } from 'types/policy.proto';
 
 import useAlertGroups from '../hooks/useAlertGroups';
 import WidgetCard from './WidgetCard';
 
-type CountsBySeverity = {
-    Low: Record<string, number>;
-    Medium: Record<string, number>;
-    High: Record<string, number>;
-    Critical: Record<string, number>;
-};
+/**
+ * This function iterates an array of AlertGroups and zeros out severities that
+ * have been filtered by the user in the widget's legend.
+ */
+function zeroOutFilteredSeverities(
+    groups: AlertGroup[],
+    hiddenSeverities: Set<PolicySeverity>
+): AlertGroup[] {
+    return groups.map(({ group, counts }) => ({
+        group,
+        counts: counts.map(({ severity, count }) => ({
+            severity,
+            count: hiddenSeverities.has(severity) ? '0' : count,
+        })),
+    }));
+}
 
-function pluckSeverityCount(severity: Severity): (group: AlertGroup) => number {
+function pluckSeverityCount(severity: PolicySeverity): (group: AlertGroup) => number {
     return ({ counts }) => {
         const severityCount = counts.find((ct) => ct.severity === severity)?.count || '0';
```

Comment:
Minor stylistic question about pro and con for falsey versus undefined values:
* `||` conditional or
* `??` nullish coalescing

Reply:
Good catch, definitely prefer `??` here. `||` is muscle memory...